docs/design/phases.md

```diff
@@ -0,0 +1,110 @@
+## Phases in kubeadm
+
+`kubeadm init` does a lot of things. In order to make kubeadm the common building block for Kubernetes we want it to be, each logical phase kubeadm does during `init` should be invokable separately. Here's an explanation what kubeadm does in the different phases:
+
+#### A note on constants / well-known values and paths
+
+Somewhere we have to draw the line what should be configurable, what shouldn't and what should be hard-coded in the binary.
+
+We've decided to make the Kubernetes directory `/etc/kubernetes` a constant in the application, since it is clearly the given path in a lot of places,
+and the most intuitive location. Having that path configurable would for sure confuse readers of a on-top-of-kubeadm-implemented deployment solution.
+
+Further, having the admin/kubelet kubeconfig path would 
+
+### Certificates
+
+First off, kubeadm generates certificate and private key pairs for different purposes.
+Certificates are stored by default in `/etc/kubernetes/pki`.
+
+There should be:
+ - a CA certificate (`ca.crt`) with its private key (`ca.key`)
+ - an API Server certificate (`apiserver.crt`) using `ca.crt` as the CA with its private key (`apiserver.key`). The certificate should:
+   - be a serving server certificate (`x509.ExtKeyUsageServerAuth`)
+   - contain altnames for
+     - the kubernetes' service's internal clusterIP and dns name (e.g. `10.96.0.1`, `kubernetes.default.svc.cluster.local`, `kubernetes.default.svc`, `kubernetes.default`, `kubernetes`)
+     - the hostname of the node
+     - the IP address of the default gateway
+     - optional extra altnames that can be specified by the user
+   - be in the `system:masters` organization
```

Comment:
That is an excellent point. We can't be too prescriptive about fields of the certs, since we need to allow certs generated by organizations to work smoothly with kubeadm.